File: 03_Data_Structures_&_Counting/10816_Frequency_Table/Answer10816.java
import java.util.HashMap;
import java.util.Scanner;

public class Answer10816 {
  public static void main(String[] args) {
    var s = new Scanner(System.in);
    s.nextLine();
    var firstList = s.nextLine().split(" ");
    s.nextLine();
    var queryList = s.nextLine().split(" ");

    var frequencyTable = new HashMap<String,Integer>();
    for(var it : firstList) {
      frequencyTable.put(it, frequencyTable.getOrDefault(it, 0) + 1);
    }
    var resultStr = new StringBuilder();
    for(int i = 0; i < queryList.length; i++) {
      if(i > 0) resultStr.append(" ");
      resultStr.append(frequencyTable.getOrDefault(queryList[i], 0));
    }
    System.out.print(resultStr);
  }
}

File: 03_Data_Structures_&_Counting/10816_Frequency_Table/Answer10816.rs
use std::io::{stdin, Read};
use std::collections::HashMap;

fn main(){
  let mut s = String::new();
  stdin().read_to_string(&mut s).unwrap();
  let inputs: Vec<&str> = s.split('\n').collect();
  let first_list:Vec<&str> = inputs[1].split_whitespace().collect();
  let query_list:Vec<&str> = inputs[3].split_whitespace().collect();

  let mut frequency_table:HashMap<&str, i32> = HashMap::new();
  for it in first_list {
    *frequency_table.entry(it).or_insert(0) += 1;
  }
  let mut result_str = String::new();
  for (i, it) in query_list.iter().enumerate() {
    if i > 0 {result_str.push(' ')}
    result_str.push_str(&frequency_table.get(*it).unwrap_or(&0).to_string());
  }
  print!("{}", result_str);
}